apps/studio/components/interfaces/DataWarehouse/TestCollectionDialog.tsx

```diff
@@ -0,0 +1,153 @@
+import { useWarehouseCollectionsQuery } from 'data/analytics/warehouse-collections-query'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+import {
+  Button,
+  CodeBlock,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogSection,
+  DialogSectionSeparator,
+  DialogTitle,
+  DialogTrigger,
+  Input,
+  TooltipContent_Shadcn_,
+  TooltipTrigger_Shadcn_,
+  Tooltip_Shadcn_,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+
+export function TestCollectionDialog({
+  accessTokens,
+  collectionToken,
+  collections,
+}: {
+  accessTokens: {
+    id: string
+    token: string
+    description?: string
+  }[]
+  collections: {
+    id: number
+    token: string
+    name: string
+  }[]
+  collectionToken: string
+  projectRef: string
+}) {
+  const BASE_WAREHOUSE_URL = `https://api.warehouse.tech/api/events`
+  const [testAccessToken, setTestAccessToken] = useState('')
+  const [selectedCollection, setSelectedCollection] = useState(collectionToken || '')
+
+  useEffect(() => {
+    setSelectedCollection(collectionToken)
+  }, [collectionToken])
+
+  useEffect(() => {
+    if (accessTokens.length > 0) {
+      setTestAccessToken(accessTokens[0].token)
+    }
+  }, [accessTokens])
+
+  const selectedAccessToken = accessTokens.find((token) => token.token === testAccessToken)
+  const selectedCollectionName = collections.find((col) => col.token === selectedCollection)?.name
+
+  return (
+    <Dialog>
+      <Tooltip_Shadcn_>
+        <TooltipTrigger_Shadcn_ asChild>
+          <DialogTrigger asChild>
+            <Button disabled={accessTokens.length === 0} type="outline">
+              Connect
+            </Button>
+          </DialogTrigger>
+        </TooltipTrigger_Shadcn_>
+        {accessTokens.length === 0 && (
+          <TooltipContent_Shadcn_>
+            Create an access token to connect to your collection
+          </TooltipContent_Shadcn_>
+        )}
+      </Tooltip_Shadcn_>
+
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Send events to this collection</DialogTitle>
+          <DialogDescription>
+            Use the following curl command to send events to this collection
+          </DialogDescription>
+        </DialogHeader>
+        <DialogSectionSeparator />
+        <DialogSection className="flex flex-col gap-4 overflow-auto">
+          <div className="flex gap-2 *:flex-1">
+            <FormItemLayout label="Collection" isReactForm={false}>
+              <Select value={selectedCollection} onValueChange={setSelectedCollection}>
+                <SelectTrigger>
+                  <span className="text-ellipsis">{selectedCollectionName || 'Collection'}</span>
+                </SelectTrigger>
+                <SelectContent>
+                  {collections.map((col) => (
+                    <SelectItem key={col.id + '-collection'} value={col.token}>
```

Comment:
an aside: you can technically do `col.id` because react keys are scoped to each component's map, i.e. adding the suffix doesn't make a difference as its impossible for clashing to happen with other renders.